hosted/aks/p1/p1_suite_test.go

```diff
@@ -361,3 +370,159 @@ func updateNodePoolModeCheck(cluster *management.Cluster, client *rancher.Client
 		return true
 	}, "5m", "5s").Should(BeTrue(), "Failed while upstream nodepool mode update")
 }
+
+// Qase ID: 221 and 292
+func updateCloudCredentialsCheck(cluster *management.Cluster, client *rancher.Client) {
+	newCCID, err := helpers.CreateCloudCredentials(client)
+	Expect(err).To(BeNil())
+	updateFunc := func(cluster *management.Cluster) {
+		cluster.AKSConfig.AzureCredentialSecret = newCCID
+	}
+	cluster, err = helper.UpdateCluster(cluster, client, updateFunc)
+	Expect(err).To(BeNil())
+	Expect(cluster.AKSConfig.AzureCredentialSecret).To(Equal(newCCID))
+	Eventually(func() bool {
+		cluster, err = client.Management.Cluster.ByID(cluster.ID)
+		Expect(err).NotTo(HaveOccurred())
+		return cluster.AKSStatus.UpstreamSpec.AzureCredentialSecret == newCCID
+	}, "5m", "5s").Should(BeTrue(), "Failed while upstream cloud credentials update")
+
+	cluster, err = helper.AddNodePool(cluster, 1, client, true, true)
+	Expect(err).To(BeNil())
+}
+
+// Qase ID: 224 and 293
+func syncAddNodePoolFromAzureAndRancher(cluster *management.Cluster, client *rancher.Client) {
+	initialNPCount := len(cluster.AKSConfig.NodePools)
+	const npAzure = "npazure"
+	By("adding nodepool from Azure", func() {
+		err := helper.AddNodePoolOnAzure(npAzure, cluster.AKSConfig.ClusterName, cluster.AKSConfig.ResourceGroup, "2")
+		Expect(err).To(BeNil())
+
+		Eventually(func() bool {
+			cluster, err = client.Management.Cluster.ByID(cluster.ID)
+			Expect(err).NotTo(HaveOccurred())
+			for _, nodePool := range cluster.AKSStatus.UpstreamSpec.NodePools {
+				if *nodePool.Name == npAzure {
+					return true
+				}
+			}
+			return false
+		}, "5m", "5s").Should(BeTrue(), "Timed out while waiting for sync from Azure")
+
+		if !helpers.IsImport {
+			// skip this check if the cluster is imported since the AKSConfig value will not be updated
+			Expect(cluster.AKSConfig.NodePools).To(HaveLen(initialNPCount + 1))
+		}
+	})
+
+	By("adding nodepool from Rancher", func() {
+		if helpers.IsImport {
+			// if the cluster is imported, update the AKSConfig value to match UpstreamSpec so that it can perform upcoming checks correctly
+			cluster.AKSConfig = cluster.AKSStatus.UpstreamSpec
+		}
+		var err error
+		cluster, err = helper.AddNodePool(cluster, 1, client, true, true)
+		Expect(err).To(BeNil())
```

Comment:
```suggestion
		Expect(err).To(BeNil())
		Expect(cluster.AKSConfig.NodePools).To(HaveLen(initialNPCount + 2))
```

Reply:
The check is already done in `AddNodePool` func.